app/src/test/java/org/dhis2/usecases/teiDashboard/dashboardfragments/notes/NotesPresenterTest.kt

```diff
@@ -0,0 +1,183 @@
+package org.dhis2.usecases.teiDashboard.dashboardfragments.notes
+
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Single
+import io.reactivex.processors.BehaviorProcessor
+import io.reactivex.processors.FlowableProcessor
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.dhis2.data.schedulers.TrampolineSchedulerProvider
+import org.dhis2.data.tuples.Pair
+import org.dhis2.usescases.teiDashboard.DashboardRepositoryImpl
+import org.dhis2.usescases.teiDashboard.dashboardfragments.notes.NotesContracts
+import org.dhis2.usescases.teiDashboard.dashboardfragments.notes.NotesPresenter
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.common.Access
+import org.hisp.dhis.android.core.common.DataAccess
+import org.hisp.dhis.android.core.enrollment.EnrollmentStatus
+import org.hisp.dhis.android.core.note.Note
+import org.hisp.dhis.android.core.note.NoteCreateProjection
+import org.hisp.dhis.android.core.program.Program
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class NotesPresenterTest {
+    private lateinit var notesPresenter: NotesPresenter
+    private val dashboardRepository: DashboardRepositoryImpl = mock()
+    private val schedulers: SchedulerProvider = TrampolineSchedulerProvider()
+    private val view: NotesContracts.View = mock()
+    private val d2: D2 = mock ()
+
+    @Before
+    fun setUp(){
+        notesPresenter = NotesPresenter(d2, dashboardRepository, schedulers, view)
+        notesPresenter.init("program_uid", "tei_uid")
+    }
+
+    @Test
+    fun `Should display message`(){
+        val message = "message"
+
+        notesPresenter.displayMessage(message)
+
+        verify(view).displayMessage(message)
+    }
+
+    @Test
+    fun `Should return true or false if program has write permission`(){
+        whenever(d2.programModule()) doReturn mock()
+        whenever(d2.programModule().programs()) doReturn mock()
+        whenever(d2.programModule().programs().uid("program_uid")) doReturn mock()
+        whenever(d2.programModule().programs().uid("program_uid")
+                .blockingGet()) doReturn getProgramDefaultAccessTrue()
+
+        assertTrue(notesPresenter.hasProgramWritePermission())
+
+        whenever(d2.programModule().programs().uid("program_uid")
+                .blockingGet()) doReturn getProgramDefaultAccessFalse()
+
+        assertTrue(!notesPresenter.hasProgramWritePermission())
+
+        whenever(d2.programModule().programs().uid("program_uid")
+                .blockingGet()) doReturn getProgramDefaultAccessNull()
+
+        assertTrue(notesPresenter.hasProgramWritePermission())
+    }
```

Comment:
A unit test should test only one thing. This could be divided in 3 separated tests for each of its case.